### ISSUE TITLE
When machine gets IPA-enrolled to different IPA with the same domain name and the same users, caches prevent correct operation

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#2501]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/2497
- Created at 2014-11-18 09:48:46 by [adelton](https://pagure.io/user/adelton)
- Closed at 2020-03-24 14:17:39 as wontfix
- Assigned to nobody
---

Recently I've created IPA + IPA-enrolled machine for testing purposes.

Then I run ipa-client-install --uninstall, created completely different IPA installation (different version even) but with the same domain and the same users.

I've IPA-enrolled the machine back but PAM authentication did not work.

Jakub says that you need to remove the sssd cache to get rid of the old uids.

This ticket is about making sure people don't hit the problem.

Possible solutions: loud warning about reusing the same domain name; removing the cache either during --uninstall or during ipa-client-install; comparing host's key timestamp in the keytab with the cache timestamp; something else.

### Comments

---

**Comment from [adelton](https://pagure.io/user/adelton) at [2014-11-21 17:26:13](https://pagure.io/SSSD/sssd/issue/2497#237789)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.14 beta

---

**Comment from [adelton](https://pagure.io/user/adelton) at [2015-08-18 12:19:20](https://pagure.io/SSSD/sssd/issue/2497#237790)**

Fields changed

rhbz:  => todo

---

**Comment from [adelton](https://pagure.io/user/adelton) at [2016-02-16 12:17:26](https://pagure.io/SSSD/sssd/issue/2497#237791)**

We need to first fix #2671, which would allow the ipa installer tools to call sss_cache --remove-db or similar. btw we also had a similar request from the realmd side.

blockedby:  => #2671
milestone: SSSD 1.14 beta => SSSD 1.14 backlog
sensitive:  => 0

---

**Comment from [adelton](https://pagure.io/user/adelton) at [2016-12-06 11:46:00](https://pagure.io/SSSD/sssd/issue/2497#237792)**

Fields changed

selected:  => May

---

**Comment from [adelton](https://pagure.io/user/adelton) at [2016-12-15 10:57:50](https://pagure.io/SSSD/sssd/issue/2497#237793)**

The manual cache removal is now implemented in sssctl. In order for this scenario to work automagically, ticket #884 must be implemented as well. In the meantime, I at least documented this in the troubleshooting guide:
https://fedorahosted.org/sssd/wiki/Troubleshooting?action=diff&version=28&old_version=27

milestone: SSSD 1.14 backlog => SSSD Patches welcome

---

**Comment from [adelton](https://pagure.io/user/adelton) at [2017-02-24 14:56:57](https://pagure.io/SSSD/sssd/issue/2497#237794)**

**Metadata Update from @adelton**:
- Issue marked as depending on: #2671
- Issue set to the milestone: SSSD Patches welcome

---

**Comment from [adelton](https://pagure.io/user/adelton) at [2020-03-24 14:17:38](https://pagure.io/SSSD/sssd/issue/2497#636848)**

Thank you for taking time to submit this request for SSSD. Unfortunately this issue was not given priority and the team lacks the capacity to work on it at this time.

Given that we are unable to fulfill this request I am closing the issue as wontfix.

If the issue still persist on recent SSSD you can request re-consideration of this decision by reopening this issue. Please provide additional technical details about its importance to you.

Thank you for understanding.

---

**Comment from [adelton](https://pagure.io/user/adelton) at [2020-03-24 14:17:40](https://pagure.io/SSSD/sssd/issue/2497#636849)**

**Metadata Update from @pbrezina**:
- Issue close_status updated to: wontfix
- Issue status updated to: Closed (was: Open)
